sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1576,8 +1576,13 @@ object CodeGenerator extends Logging {
    * they are explicitly removed. A Cache on the other hand is generally configured to evict entries
    * automatically, in order to constrain its memory footprint.  Note that this cache does not use
    * weak keys/values and thus does not respond to memory pressure.
+   *
+   * We wrap Guava Cache with NonFateSharingCache: tasks for similar queries that run concurrently
+   * can hit the same codegen cache key at the same time. If one query fails or gets canceled while
+   * some of its tasks are doing codegen, it will cause spurious failures in all other queries with
+   * tasks that wait on the same cache key. See NonFateSharingCache for more details.
```

Comment:
If we move some of this to the NonFateSharingCache doc comment, it might suffice to say something simpler here, like:
> Codegen can be slow. Use a non fate sharing cache in case a query gets canceled during code gen while other queries wait on the same code, so that those other queries don't get wrongly aborted. See [[NonFateSharingCache]] for more details.